server/sonar-search/src/main/java/org/sonar/search/SearchServer.java

```diff
@@ -45,7 +50,14 @@ public SearchServer(Props props) {
   public void start() {
     Jmx.register(EsSettingsMBean.OBJECT_NAME, settings);
     initBootstrap();
-    node = NodeBuilder.nodeBuilder().settings(settings.build()).build();
+    Settings esSettings = settings.build();
+    if (esSettings.getAsInt("discovery.zen.minimum_master_nodes", 1) >= 2) {
```

Comment:
"discovery.zen.minimum_master_nodes" should be defined in a constant because it's used 3 times in this class.